baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/Slf4jLogsafeArgs.java

```diff
@@ -68,24 +73,27 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
         }
 
         List<? extends ExpressionTree> allArgs = tree.getArguments();
-        int lastIndex = allArgs.size() - 1;
-        int startArg = MARKER.matches(allArgs.get(0), state) ? 2 : 1;
-        ExpressionTree lastArg = allArgs.get(lastIndex);
-        boolean lastArgIsThrowable = THROWABLE.matches(lastArg, state);
-        int endArg = lastArgIsThrowable ? lastIndex - 1 : lastIndex;
+        int startArgIndex = MARKER.matches(allArgs.get(0), state) ? 2 : 1;
+        int lastArgIndex = allArgs.size() - 1;
+        ExpressionTree lastArg = allArgs.get(lastArgIndex);
 
-        ImmutableList.Builder<Integer> badArgsBuilder = ImmutableList.builder();
-        for (int i = startArg; i <= endArg; i++) {
-            if (!ARG.matches(allArgs.get(i), state)) {
-                badArgsBuilder.add(i);
-            }
+        if (startArgIndex == lastArgIndex && OBJECT_ARRAY.matches(lastArg, state)) {
+            return Description.NO_MATCH;
         }
-        List<Integer> badArgs = badArgsBuilder.build();
-        if (badArgs.isEmpty() || TestCheckUtils.isTestCode(state)) {
+
+        boolean lastArgIsThrowable = THROWABLE.matches(lastArg, state);
+        int lastNonThrowableArgIndex = lastArgIsThrowable ? lastArgIndex - 1 : lastArgIndex;
+
+        List<Integer> badArgIndices = IntStream.rangeClosed(startArgIndex, lastNonThrowableArgIndex)
+                .filter(i -> !ARG.matches(allArgs.get(i), state))
+                .boxed()
+                .collect(Collectors.toList());
```

Comment:
Any chance we can avoid the stream here? This code is _super_ hot because we log a lot of things, one large internal project in particular benefits from any compile speedup we can provide!

Reply:
Oh, I thought I spotted a nice simplification. Not my intention to slow it down. I'll revert to the previous for-each loop.

Reply:
Reverted but had to factor it out to a method to pass the cyclomatic check.